lea/runner.py

```diff
@@ -368,12 +440,17 @@ def get_status(view_key):
                             ),
                         )
                     else:
-                        job = functools.partial(
-                            self.client.materialize_view,
-                            view=self.views[view_key].with_context(
-                                table_reference_mapping=table_reference_mapping
-                            ),
+                        view = self.views[view_key].with_context(
+                            table_reference_mapping=table_reference_mapping
                         )
+                        if materialize_ctes and isinstance(view, SQLView):
+                            print(self._materialize_ctes_and_view)
+                            job = functools.partial(self._materialize_ctes_and_view, view=view)
+                        else:
+                            job = functools.partial(
+                                self.client.materialize_view,
+                                view=view,
+                            )
```

Comment:
I wouldn't approach it this way. What I would suggest, is to split the query into CTEs, and create a `View` object for each CTE. I would then add these objects to the list of views that will be executed. This way, you benefit from the rest of lea, and don't need to handle the execution adhoc. I think you should rework your code so that it leverages what already exists, if that makes sense.